core/message-tags-3.3.md

```diff
@@ -80,6 +88,39 @@ Individual tag keys MUST only be used a maximum of once per message. Clients
 receiving messages with more than one occurrence of a tag key SHOULD discard all
 but the final occurrence.
 
+### The `TAGMSG` tag-only event
+
+A new event `TAGMSG` is defined for sending messages with tags but no text content.
+This event MUST be delivered to targets in the same way as `PRIVMSG` and `NOTICE`
+events. This means for example, honouring channel membership, modes,
+[`echo-message`](../extensions/echo-message-3.2.html),
+[`STATUSMSG`](https://tools.ietf.org/html/draft-hardy-irc-isupport-00#section-4.18)
+prefixes, etc.
+
+Servers MAY apply moderation to this event using existing or newly specified modes or
+configuration.
+
+These events MUST NOT be delivered to clients that haven't negotiated the message tags
+capability and MAY be rejected if no tags are included.
+
+If this event is sent without any tags, servers SHOULD reject it with a standard
```

Comment:
Not that important, but any reason this is not a MUST? i.e. if servers choose to disallow tagless TAGMSG then they must use the given numeric for that. As currently stands servers are allowed to do whatever they want making this a useless requirement.